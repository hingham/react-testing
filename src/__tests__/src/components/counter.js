import React from "react";

import renderer from "react-test-renderer";
import Counter from "../../../components/counter/counter.js";

describe("<Counter />", () => {
  it("is alive", () => {
    expect(true).toBeTruthy;
  });

  it("changes state on a click", () => {
    let component = mount(<Counter />);
    let link = component.find("#right");
    link.simulate("click");
    expect(component.state("count")).toBe(1);
    expect(component.find("span").text()).toContain("1");
  });

  it("decrements number", () => {
    let component = mount(<Counter />);
    let link = component.find("#left");
    link.simulate("click");
    expect(component.state("count")).toBe(-1);
    expect(component.find("span").text()).toContain("1");
  });

  it('renders correctly', ()=>{
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
    })

});
