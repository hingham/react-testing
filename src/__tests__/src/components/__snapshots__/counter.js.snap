// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section
  className="counter centered"
>
  <a
    className="down clicker centered"
    href="#"
    id="left"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <p>
    hello
  </p>
  <a
    className="up clicker centered"
    href="#"
    id="right"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
